### PR TITLE
Docs migration to Docusaurus

### PR DESCRIPTION
**Pull request by nunix**
Docusaurus installed and initial configuration done:
- Elemental sidebar created to match the mkdocs side bar.
- Header links set and logo is missing
- Footer created with same links from mkdocs

The docs have been migrated 1:1 without any adaptation, so expect view glitches.

### CODE REVIEW

**Comment by nunix** [comment]
Commit [9f0204f](https://github.com/rancher/elemental/pull/272/commits/9f0204f9db2daead90d61fc4157d50b9d251e9e8) also added:

- Plugin `raw-loader` has been added for loading YAML and JSON files content
- `CodeBlock` module has been added as a Global module (same as Tabs and TabItem)

**Comment by kkaempf** [comment]
No update from @nunix - moving to blocked

**Review by kkaempf — CHANGES_REQUESTED** [review]
I don't think `.docusaurus` should be under git control as it contains generated files.

**Comment by kkaempf** [comment]
Please also consider https://github.com/nunix/elemental-docs/pull/4 and https://github.com/nunix/elemental-docs/pull/5

**Comment by kkaempf** [comment]
The files in `img` need a LICENSE and probably author attributions.

**Comment by kkaempf** [comment]
`src/theme` seems to come from upstream (?) and shouldn't be checked into git. If we can't pull these files at buildtime, they need a license and proper attribution.

**Comment by kkaempf** [comment]
I also wonder if we shouldn't take this move to docusaurus as an opportunity to migrate all docs (elemental, elemental-operator, elemental-toolkit) into a single, new repository `elemental-docs` 🤔

**Comment by nunix** [comment]
> `src/theme` seems to come from upstream (?) and shouldn't be checked into git. If we can't pull these files at buildtime, they need a license and proper attribution.

@kkaempf this directory is needed for the global variables, it's indeed Docusaurus content, but adapted to Elemental.

**Comment by kkaempf** [comment]
> > `src/theme` seems to come from upstream (?) and shouldn't be checked into git. If we can't pull these files at buildtime, they need a license and proper attribution.
> 
> @kkaempf this directory is needed for the global variables, it's indeed Docusaurus content, but adapted to Elemental.

If it's coming from Docusaurus, it needs to me mentioned in a (toplevel?) license file with proper attribution. What does Docusaurus say about including theme files ?

**Review by kkaempf — CHANGES_REQUESTED** [review]
The docusaurus content (theme and `png`s) needs proper licensing and attribution

**Comment by kkaempf** [comment]
Merging this will immediately break docs creation on https://rancher.github.io/elemental, no ?

**Comment by nunix** [comment]
> > > `src/theme` seems to come from upstream (?) and shouldn't be checked into git. If we can't pull these files at buildtime, they need a license and proper attribution.
> > 
> > 
> > @kkaempf this directory is needed for the global variables, it's indeed Docusaurus content, but adapted to Elemental.
> 
> If it's coming from Docusaurus, it needs to me mentioned in a (toplevel?) license file with proper attribution. What does Docusaurus say about including theme files ?

Docusaurus has an MIT license (https://github.com/rancher/elemental/pull/272#issuecomment-1294709443) and concerning the theme files, this is part of the product itself and usable by the end-users.

More precisely, concerning the directory above, it has been created using a Docusaurus features: https://docusaurus.io/docs/swizzling and it's even encouraged in their own docs: https://docusaurus.io/docs/markdown-features/react#mdx-component-scope.

**Comment by nunix** [comment]
> Merging this will immediately break docs creation on https://rancher.github.io/elemental, no ?

yes, the GH action still needs to be adapted for Docusaurus (I'll check with someone in the team)

**Review by Itxaka — APPROVED** [review]

File: docs/cluster-reference.md [hunk 1]
```diff
@@ -55,52 +64,52 @@ A list of `#!yaml machinePools`. A minimun of 1 `#!yaml machinePools` is require
...
+A `machineConfigRef` is a generic k8s `ObjectReference` which usually contain a
+`kind` `name` and `apiVersion` to point to a different object.
 
 In Elemental, we set this to a [MachineInventorySelectorTemplate](machineinventoryselectortemplate-reference.md).
 This allows us to point to more than one object by using the selector.
 
-
 ### Example
 
+The example below creates a cluster that uses 2 different `machinePools` to set different nodes to control-plane and workers nodes,
```
Comment (kkaempf) [inline docs/cluster-reference.md:106]:
```suggestion
The example below creates a cluster that uses 2 different `machinePool`s to set different nodes to control-plane and workers nodes,
```

**Review by kkaempf — CHANGES_REQUESTED** [review]

**Review by mjura — APPROVED** [review]
LGTM

**Review by kkaempf — APPROVED** [review]
Ship it !

**Pull request merged**